src/m365/spo/commands/applicationcustomizer/applicationcustomizer-remove.spec.ts

```diff
@@ -0,0 +1,363 @@
+import Command, { CommandError } from '../../../../Command';
+import commands from '../../commands';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import auth from '../../../../Auth';
+import { telemetry } from '../../../../telemetry';
+import { pid } from '../../../../utils/pid';
+import { session } from '../../../../utils/session';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Cli } from '../../../../cli/Cli';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import request from '../../../../request';
+import { Logger } from '../../../../cli/Logger';
+
+const command: Command = require('./applicationcustomizer-remove');
+
+describe(commands.APPLICATIONCUSTOMIZER_REMOVE, () => {
+  let commandInfo: CommandInfo;
+  const webUrl = 'https://contoso.sharepoint.com';
+  const id = '14125658-a9bc-4ddf-9c75-1b5767c9a337';
+  let promptOptions: any;
+  let log: any[];
+  let logger: Logger;
+  let requests: any[];
+
+  const defaultDeleteCallsStub = (): sinon.SinonStub => {
+    return sinon.stub(request, 'delete').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/Web/UserCustomActions(') > -1) {
+        return Promise.resolve(undefined);
+      }
+
+      if ((opts.url as string).indexOf('/_api/Site/UserCustomActions(') > -1) {
+        return Promise.resolve(undefined);
+      }
+
+      return Promise.reject('Invalid request');
+    });
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    requests = [];
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.delete,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId
+    ]);
```

Comment:
Hi, just popping in here to say that we should refactor this `sinonUtil.restore([...` to `sinon.restore()`. This is a new way to ensure that the entire sandbox is cleared and will from now on be used in every command.

Reply:
I am converting it to a draft. I will make the suggested changes and mark it as ready for review.